src/Middleware/Configuration/Transaction.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Messenger\Middleware\Configuration;
+
+use Symfony\Component\Messenger\EnvelopeItemInterface;
+
+/**
+ * Marker item to tell this message should be handled in a different transaction.
```

Comment:
A longer explanation of the idea could be great in here, as "transaction" might confuse :)